### PR TITLE
TLA+ spec of CCV

### PR DESCRIPTION
**Pull request by Kukovec**
This PR ports and updates the previously created CCV TLA+ spec from https://github.com/cosmos/interchain-security/tree/jk/ISaudit.
It includes additional modeling of provider-side timeout.

### CODE REVIEW

**Review by p-offtermatt — COMMENTED** [review]
Looks good, mostly just nitpicks.
My only bigger issue is that the document the invariants were taken from
was deprecated. I think those should be properties outlined in the spec instead.

File: spec/app/ics-028-cross-chain-validation/tla/CCV.tla [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+    /\ votingPowerRunning' \in [newValidators -> Nat]
+    /\ \A v \in newValidators: votingPowerRunning'[v] > 0
+    \* Note: votingPowerHasChanged' is set to true 
+    \* even if votingPowerRunning' = votingPowerRunning
+    /\ votingPowerHasChanged' = TRUE
+    /\ UNCHANGED consumerVars
+    /\ UNCHANGED << votingPowerHist, expectedResponders, maturePackets >>
+    /\ lastAction' = "VotingPowerChange"
+
+RcvPacket ==
```
[the remark this answers is not in the document]
Comment (mpoke) [reply spec/app/ics-028-cross-chain-validation/tla/CCV.tla:276]: RcvVPCPacket and make it consistent throughout the entire model.

**Comment by p-offtermatt** [comment]
@mpoke applied the review suggestions.
Decided to leave the invariants from the old QA document (but the link is fixed),
since it is unclear how we will proceed with it.

**Review by p-offtermatt — APPROVED** [review]

**Review by mpoke — CHANGES_REQUESTED** [review]
@p-offtermatt here are some comments after a first review. I still have to review the properties and invariants.

**Comment by crodriguezvega** [comment]
As discussed we will close this PR and add a link to the existing Quint spec.

**Pull request closed without merging**